lib/rubocop/cop/utils/regexp_ranges.rb

```diff
@@ -6,27 +6,41 @@ module Utils
       # Helper to abstract complexity of building range pairs
       # with octal escape reconstruction (needed for regexp_parser < 2.7).
       class RegexpRanges
-        attr_reader :compound_token, :root
+        attr_reader :root
 
         def initialize(root)
           @root = root
-          @compound_token = []
+          @populated = false
+        end
+
+        def compound_token
+          populate_all unless @populated
+
+          @compound_token
         end
 
         def pairs
-          unless @pairs
-            @pairs = []
-            populate(root)
-          end
+          populate_all unless @populated
+
+          @pairs
+        end
+
+        private
+
+        def populate_all
+          @compound_token = []
+          @pairs = []
```

Comment:
It looks like the initialization of these instance variables can be moved to the `initialize` method.

Reply:
I intentionally moved this to `#populate_all` since this method is responsible for lifecycle of `@compound_token` and `@pairs`. For instance existing and new code are NOT thread-safe - if we want to add a lock then it should be in this method with initializing this value anyway.

If you still prefer to put them in `#initialize` just let me know - I'll move it with a comment

Reply:
Yeah, I find it easier to understand when these instance variables are set up and prepared at the point when object is instantiated. Can you move these into `initialize`?

Reply:
Moved!